### PR TITLE
More UI Enhancements for Ongoing Elections

### PR DESCRIPTION
**Pull request by brandonfancher**
## In this PR

- [x] Add all participants to previous election rounds
- [x] Add examples of election video upload play buttons
- [x] Add round timer UI

## UI Goals

image

## Later (in a different PR)
* Consens-o-meter

### CODE REVIEW

File: packages/webapp/src/_app/hooks/utils.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    useEffect(() => {
+        const tick = () => {
+            savedCallback.current();
+        };
+        if (delay !== null) {
+            const id = setInterval(tick, delay);
+            return () => clearInterval(id);
+        }
+    }, [delay]);
+};
```
Comment (cool-ant) [inline packages/webapp/src/_app/hooks/utils.ts:21]: Really nice.

Comment (brandonfancher) [reply packages/webapp/src/_app/hooks/utils.ts:21]: Thanks! (I stole it from the internet.)

File: packages/webapp/src/_app/ui/pie-chart.tsx [hunk 2]
```diff
@@ -0,0 +1,57 @@
+interface Props {
+    percent: number;
+    size: number;
+    backgroundColor?: string;
+    fillColor?: string;
+}
+
+export const PieChart = ({
```
Comment (cool-ant) [inline packages/webapp/src/_app/ui/pie-chart.tsx:8]: I'm being a bit picky, but this isn't a pie *chart* (or at least isn't much of one). I'd like to name it a bit more accurately. Either call it a pie-status-indicator or pie-countdown-clock. or a countdown pie-mer. lol. no idea, but a name that wouldn't conjure pictures of multiple differently-colored segments

File: packages/webapp/src/elections/components/countdown.tsx [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+            <PieChart percent={percent} size={28} />
+            <div className="flex-1 ml-2">
+                <Text size="sm" className="font-semibold">
+                    Vote Timer
+                </Text>
+                <Text size="sm">{timeRemainingString}</Text>
+            </div>
+        </div>
+    );
+};
```
Comment (cool-ant) [inline packages/webapp/src/elections/components/countdown.tsx:46]: I'm liking Pie-mer. :)

File: packages/webapp/src/elections/components/election-member-chips.tsx [hunk 4]
```diff
@@ -72,7 +83,23 @@ interface DelegateChipProps {
...
+    member: MemberData;
+    delegateLevel?: string;
+    isDelegate?: boolean;
+    electionVideoCid?: string;
+}
+
+export const ElectionParticipantChip = ({
+    member,
+    delegateLevel,
+    isDelegate = false,
```
Comment (cool-ant) [inline packages/webapp/src/elections/components/election-member-chips.tsx:100]: isDelegate is minimally used. Is there a future use for it? Or could you just default delegateLevel = 0 and use that as the flag in the code?

**Review by cool-ant — APPROVED** [review]

File: packages/webapp/src/elections/components/countdown.tsx (hunk 3 above)
Comment (brandonfancher) [reply packages/webapp/src/elections/components/countdown.tsx:46]: ✅

File: packages/webapp/src/_app/ui/pie-chart.tsx (hunk 2 above)
Comment (brandonfancher) [reply packages/webapp/src/_app/ui/pie-chart.tsx:8]: Yeah, I hear you. This component is not the countdown and has nothing to do with time, but it's not a full-fledged pie chart either.

File: packages/webapp/src/elections/components/election-member-chips.tsx (hunk 4 above)
Comment (brandonfancher) [reply packages/webapp/src/elections/components/election-member-chips.tsx:100]: Yeah, in fact, if we look at `GenericMemberChip` and `ElectionParticipantChip`, I hint with TODOs on that prop is likely to change or go away depending on the shape of the data coming in and whether we can simply rely on that. This was an easy, in-your-face, way to demonstrate the application of the delegate badge...but we'll likely get that from the member object when we hook it up.

**Pull request merged**